### PR TITLE
MINIFICPP-1624 Allow building in docker without up-to-date compilers …

### PR DESCRIPTION
**Pull request by szaszm**
…on the host

https://issues.apache.org/jira/browse/MINIFICPP-1624

After MINIFICPP-1425, there is no reason to require C++20-compatible compilers on hosts that are only used to launch docker containers for building. Please note that cmake still needs some compiler and generator target (e.g. make or ninja) to be installed on the host to be able to generate a build system.

---

Thank you for submitting a contribution to Apache NiFi - MiNiFi C++.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced
     in the commit message?

- [x] Does your PR title start with MINIFICPP-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [x] Has your PR been rebased against the latest commit within the target branch (typically main)?

- [x] Is your initial contribution a single, squashed commit?

### For code changes:
- [x] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [x] If applicable, have you updated the LICENSE file?
- [x] If applicable, have you updated the NOTICE file?

### For documentation related changes:
- [x] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check GitHub Actions CI results for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -428,23 +460,17 @@ if (ENABLE_ALL OR ENABLE_COAP STREQUAL "ON")
...
-		include(BundledPugiXml)
-		use_bundled_pugixml(${CMAKE_CURRENT_SOURCE_DIR} ${CMAKE_CURRENT_BINARY_DIR})
-		createExtension(WEL-EXTENSION "WEL EXTENSIONS" "Enables the suite of Windows Event Log extensions." "extensions/windows-event-log" "extensions/windows-event-log/tests")
-	endif()
-endif(WIN32)
+if (WIN32 AND (ENABLE_ALL OR ENABLE_WEL))
+	include(BundledPugiXml)
+	use_bundled_pugixml(${CMAKE_CURRENT_SOURCE_DIR} ${CMAKE_CURRENT_BINARY_DIR})
+	createExtension(WEL-EXTENSION "WEL EXTENSIONS" "Enables the suite of Windows Event Log extensions." "extensions/windows-event-log" "extensions/windows-event-log/tests")
+endif(WIN32 AND (ENABLE_ALL OR ENABLE_WEL))
```
Comment (fgerlits) [inline CMakeLists.txt:467]: I would change this to `endif()`

File: CMakeLists.txt [hunk 2]
```diff
@@ -96,6 +83,59 @@ set_property(CACHE STRICT_GSL_CHECKS PROPERTY STRINGS ${STRICT_GSL_CHECKS_Values
...
+option(ENABLE_GPS "Enables the GPS extension." OFF)
+option(ENABLE_COAP "Enables the CoAP extension." OFF)
+option(ENABLE_SQL "Enables the SQL Suite of Tools." OFF)
+option(ENABLE_MQTT "Enables the mqtt extension." OFF)
+option(ENABLE_PCAP "Enables the PCAP extension." OFF)
+option(ENABLE_LIBRDKAFKA "Enables the librdkafka extension." OFF)
+option(DISABLE_SCRIPTING "Disables the scripting extensions." OFF)
+option(ENABLE_SENSORS "Enables the Sensors package." OFF)
+option(ENABLE_USB_CAMERA "Enables USB camera support." OFF)
+option(ENABLE_TENSORFLOW "Disables the TensorFlow extensions." OFF)  ## Disabled by default because TF can be complex/environment-specific to build
```
Comment (fgerlits) [inline CMakeLists.txt:109]: old typo, but should be "Enables ..."

File: CMakeLists.txt [hunk 3]
```diff
@@ -96,6 +83,59 @@ set_property(CACHE STRICT_GSL_CHECKS PROPERTY STRINGS ${STRICT_GSL_CHECKS_Values
 
 if (WIN32)
 	option(MSI_REDISTRIBUTE_UCRT_NONASL "Redistribute Universal C Runtime DLLs with the MSI generated by CPack. The resulting MSI is not distributable under Apache 2.0." OFF)
+	option(ENABLE_WEL "Enables the suite of Windows Event Log extensions." OFF)
+	option(ENABLE_PDH "Enables PDH support." OFF)
+endif()
+
+if(CMAKE_SYSTEM_NAME STREQUAL "Linux")
+	option(ENABLE_SYSTEMD "Disables the systemd extension." ON)
```
Comment (fgerlits) [inline CMakeLists.txt:91]: the description should say "Enables ..."

File: CMakeLists.txt (hunk 1 above)
Comment (szaszm) [reply CMakeLists.txt]: Changed in eaf13e4

File: CMakeLists.txt (hunk 2 above)
Comment (szaszm) [reply CMakeLists.txt]: fixed in eaf13e4

File: CMakeLists.txt (hunk 3 above)
Comment (szaszm) [reply CMakeLists.txt]: fixed in eaf13e4

**Review by lordgamez — APPROVED** [review]

**Review by fgerlits — APPROVED** [review]

**Review by arpadboda — APPROVED** [review]

**Pull request closed without merging**